includes/SiteGen/SiteGen.php

```diff
@@ -70,11 +70,21 @@ private static function check_capabilities() {
 	 * @param string $site_description The site description
 	 */
 	private static function get_refined_site_description( $site_description ) {
```

Comment:
Why are we jumping through all these hoops to transform arrays and not just enforcing a string at the parameter level via type declaration?

Reply:
@earnjam   
It is because we combine a few things for prompts.  
Some calls may need more than just site\_description, that was the initial thought, that we will use output of some calls to generate others.  
It is easier to handle the usecase here and convert and give back the site description than changing where we call the function and look for any potential changes introduced...

Reply:
But site description is a string. This function returns a string (or should be). So we should be passing a string as the parameter. 

Instead of juggling arrays, we should figure out why there is a scenario why a string wouldn't be what is passed. 

The fact that it isn't a string being passed is the bug, not that the function can't handle unexpected types.

Reply:
Ok so site\_description can a misleading name here, it isn't always a string, the idea of these changes is to make sure that case is handled, down the file there are changes which pass the array directly and the function handles it to just update the site description. So an array is not an "unexpected type" per se.